docs/DeveloperGuide.md

```diff
@@ -225,6 +225,59 @@ Aspect: How navigation between directory works
   - Pros: Easy to implement since we just have to return a new CommandResult which has an additional parameter of Directory.
   - Cons: Break the abstraction layer as Commands (Logic) should not have to be aware of how the Model is working.
 
+### Editing Information
+
+#### Implementation
+The proposed editing mechanism is facilitated by `EditDescriptor`, `Index` and `EditCommand` which extends `Command`.
+Editing can be done to travelplan objects such as `Friend`, `Accommodation` and `Activity` or to `TravelPlan`.
+
+`EditCommand` uses `Index` to identify the object to be modified.
+`Index` provide standardisation between the components to refer to a specific travel plan object or travelplan.
+
+`EditCommand` uses `EditDescriptor` to create an edited object.
+`EditDescriptor` provides method `buildFromSource(source)` to build a new object based fields specified within the user input.
+
+Each object has a specific set of valid fields that can be modified. 
+- `Activity`: Name, Importance, Cost, DateTime, Location 
+- `Accommodation`: Name, StartDate, EndDate, Location, Cost
+- `Friend`: Name, Passport, Mobile
+- `TravelPlan`: Name, StartDate, EndDate
+
+EditCommand must have at least one field that is edited. Any fields that are invalid will throw an error. 
+Hence, child classes of EditCommand accounts for editing valid types of field. Each of the following classes extends `EditCommand`
+- `EditActivityCommand`
+- `EditAccommodationCommand`
+- `EditFriendCommand`
+- `EditTravelPlanCommand`
+
+`EditCommand` accounts for duplicated objects.
+Two objects are considered the same when they have the same name.
```

Comment:
I think the accomodation class requires start date and end date to be the same too for two accommodation object to be the same.

Reply:
yup, travelplan too. I updated them